pkg/rest/rest.go

```diff
@@ -112,55 +132,15 @@ type VirtualMachineStateHandler interface {
 	SetVMState(c *gin.Context)
 }
 
-// parseRestfulURI validates the input URI and returns an URL object
-func parseRestfulURI(inputURI string) (*url.URL, error) {
-	restURI, err := url.ParseRequestURI(inputURI)
-	if err != nil {
-		return nil, err
-	}
-	scheme, err := toRestScheme(restURI.Scheme)
-	if err != nil {
-		return nil, err
-	}
-	if scheme == TCP && len(restURI.Host) < 1 {
-		return nil, errors.New("invalid TCP uri: missing host")
-	}
-	if scheme == TCP && len(restURI.Path) > 0 {
-		return nil, errors.New("invalid TCP uri: path is forbidden")
-	}
-	if scheme == TCP && restURI.Port() == "" {
-		return nil, errors.New("invalid TCP uri: missing port")
-	}
-	if scheme == Unix && len(restURI.Path) < 1 {
-		return nil, errors.New("invalid unix uri: missing path")
-	}
-	if scheme == Unix && len(restURI.Host) > 0 {
-		return nil, errors.New("invalid unix uri: host is forbidden")
-	}
-	if scheme == Unix && len(restURI.Path) > maxSocketPathLen() {
-		return nil, fmt.Errorf("invalid unix uri: socket path length exceeds macOS limits")
-	}
-	return restURI, err
-}
-
-// toRestScheme converts a string to a ServiceScheme
+// toRestScheme converts a string to a [ServiceScheme].
 func toRestScheme(s string) (ServiceScheme, error) {
-	switch strings.ToUpper(s) {
```

Comment:
This is to ensure no mistakes are made when this is specified manually.
Either use `upper` or `lower`

Reply:
Can you point me out to a case in the code where it is specified manually?

From what I see, it is only used to parse a Scheme field from a structure obtained from url.ParseRequestURI, which ensures Scheme is lowercase, so we can expect the lowercased string here. This was noted in the commit message, too.

Reply:
I'll let @cfergeau decide as he maintains this codebase. 

I just notice that https://github.com/crc-org/vfkit/pull/292/files#diff-e8eb5f08837c85d4c2db17fa463051c4a787efb7cbc9883092f53b24a48514e1R32 uses the input that comes from the request parse; and it assumes the format to be returned always to be lower. Defensive programming is not wrong, as this is not code to be performant or such (and mistakes have been made before). Overall, this looks more maintainable.